internal/handlers/telegram/handler_test.go

```diff
@@ -10,6 +10,160 @@ import (
 	"testing"
 )
 
+func MockUpdateHandler(updates tgbotapi.UpdatesChannel) string {
+	for u := range updates {
+		switch {
+		case u.Message == nil:
+			return "Missing message data"
+		case u.Message.NewChatMembers != nil:
+			return "joinHandler triggered"
+		case u.Message.LeftChatMember != nil:
+			return "partHandler triggered"
+		case u.Message.Text != "":
+			return "messageHandler triggered"
+		case u.Message.Sticker != nil:
+			return "stickerHandler triggered"
+		case u.Message.Document != nil:
+			return "documentHandler triggered"
+		case u.Message.Photo != nil:
+			return "photoHandler triggered"
+		default:
+			return "Triggered, but message type is currently unsupported"
+		}
+	}
+	return ""
+}
+
+func TestJoinUpdate(t *testing.T) {
+
+	updates := make(chan tgbotapi.Update, 1)
+	update := tgbotapi.Update {
+			Message: &tgbotapi.Message{
+				NewChatMembers: &[]tgbotapi.User{
+					tgbotapi.User{
+						ID:        1,
+						FirstName: "test",
+						UserName:  "testUser",
+					},
+				},
+			},
+		}
+	updates <- update
+
+	correct := "joinHandler triggered"
+	assert.Equal(t, correct, MockUpdateHandler(updates))
+
+	// clientObj := &Client{
+	// 	IRCSettings: &internal.IRCSettings{
+	// 		Prefix:          "<",
+	// 		Suffix:          ">",
+	// 		ShowJoinMessage: true,
+	// 	},
+	// 	logger: internal.Debug {
+	// 		DebugLevel: true,
+	// 	},
+	// 	sendToIrc: func(s string) {
+	// 	},
+	// }
+	// updateHandler(clientObj, updates)
```

Comment:
Calling the update handler takes long time, and does not even complete when I run `go test ./...`. Your input on the reason is welcome, but I believe it stays in the loop waiting for another update.

Reply:
Most likely due to waiting for update events.

Reply:
@Zedjones do you know why the client would be stalling on the channel here?